internal/pkg/plugin/definition.go

```diff
@@ -33,12 +41,24 @@ type SharingPlugin interface {
 
 	// Prepare the sharing plugins, such as build the connection with the external services.
 	Prepare() error
+	// Start a server to receive the input APM data.
+	Start() error
 	// Close the sharing plugin.
 	Close() error
 }
 
 // Config is used to initialize the DefaultInitializingPlugin.
 type Config map[string]interface{}
 
-// NameField is a required field in Config.
-const NameField = "plugin_name"
+// GetPluginName returns a plugin name generated by the packages.
+func GetPluginName(p Plugin) string {
```

Comment:
why not just return the string name?

Reply:
Use the directory structure to ensure that the plug-in name is unique